### PR TITLE
MRG: Add add_noise to simulation

### PR DESCRIPTION
**Pull request by larsoner**
One part of what we need for https://github.com/mne-tools/mne-python/issues/5058#issuecomment-377500991

### CODE REVIEW

File: mne/simulation/evoked.py [hunk 1]
```diff
@@ -107,9 +107,65 @@ def simulate_noise_evoked(evoked, cov, iir_filter=None, random_state=None):
...
+    cov : instance of Covariance
+        The noise covariance.
+    iir_filter : None | array-like
+        IIR filter coefficients (denominator).
+    random_state : None | int | np.random.RandomState
+        To specify the random generator state.
+
+    Returns
+    -------
+    noise : evoked object
```
Comment (agramfort) [inline mne/simulation/evoked.py]: fix

File: mne/simulation/evoked.py [hunk 2]
```diff
@@ -107,9 +107,65 @@ def simulate_noise_evoked(evoked, cov, iir_filter=None, random_state=None):
     .. versionadded:: 0.10.0
     """
     noise = evoked.copy()
-    noise.data = _generate_noise(evoked.info, cov, iir_filter, random_state,
-                                 evoked.data.shape[1])[0]
-    return noise
+    noise.data[:] = 0
+    return _add_noise(noise, cov, iir_filter, random_state)
```
Comment (agramfort) [inline mne/simulation/evoked.py]: I would suggest to deprecate then simulate_noise_evoked as it's purpose was to add noise after simulating evoked data

**Comment by larsoner** [comment]
@agramfort done

**Comment by agramfort** [comment]
ok for you @makkostya and @ngayraud ?

**Comment by agramfort** [comment]
ping @makkostya  @ngayraud @sdeslauriers 

ok for you?

**Comment by ngayraud** [comment]
Looks good!

**Comment by agramfort** [comment]
great

ok for you too @makkostya ?

**Comment by makkostya** [comment]
If I understood well, the function is replacing the inst.data by simulated noise, not adding it?
Thus adding noise is done outside the function (something like soi.data + noise.data)?
If it is like this, I'm Ok with it. 

Just the line:
`for epoch in data: `
will it work properly if inst is a raw?

File: mne/simulation/evoked.py [hunk 3]
```diff
@@ -106,10 +109,75 @@ def simulate_noise_evoked(evoked, cov, iir_filter=None, random_state=None):
...
+@verbose
+def add_noise(inst, cov, iir_filter=None, random_state=None,
+              verbose=None):
+    """Create noise as a multivariate Gaussian.
+
+    The spatial covariance of the noise is given from the cov matrix.
+
+    Parameters
+    ----------
+    inst : instance of Evoked, Epochs, or Raw
```
Comment (agramfort) [inline mne/simulation/evoked.py]: @larsoner I don't see any test for adding noise to evoked. Is the docstring wrong? or is test missing?

**Comment by larsoner** [comment]
> If I understood well, the function is replacing the inst.data by simulated noise, not adding it?

Oops, that's a bug.

> will it work properly if inst is a raw?

Yes internally the function makes a new view of the data array, i.e. shape `(1, 306, ...)` if it's raw so it's unified.

File: mne/simulation/evoked.py (hunk 3 above)
Comment (larsoner) [reply mne/simulation/evoked.py]: No test for that, I can add it

**Comment by agramfort** [comment]
> Oops, that's a bug.

good catch then :) test seems not good enough :-)

**Comment by larsoner** [comment]
Agreed :)

**Comment by ngayraud** [comment]
@makkostya good catch!

**Comment by agramfort** [comment]
thx @larsoner and @makkostya / @ngayraud for taking a look.

it would be great to have a tutorial soon on how to simulate data.

**Pull request merged**